ui/pages/institutional/confirm-add-custodian-token/confirm-add-custodian-token.js

```diff
@@ -56,6 +58,71 @@ const ConfirmAddCustodianToken = () => {
     }
   }, [connectRequest, history, mostRecentOverviewPage]);
 
+  const handleButtonClick = useCallback(
+    async (isConfirm) => {
+      try {
+        if (isConfirm) {
+          setConnectError('');
+          setIsLoading(true);
+
+          if (connectRequest.chainId) {
+            const networkType = Object.keys(BUILT_IN_NETWORKS).find(
+              (key) =>
+                Number(BUILT_IN_NETWORKS[key].chainId).toString(10) ===
+                connectRequest.chainId.toString(),
+            );
+
+            await dispatch(setProviderType(networkType));
+          }
+
+          let custodianName = connectRequest.service.toLowerCase();
+
+          if (connectRequest.service === 'JSONRPC') {
+            custodianName = connectRequest.environment;
+          }
+
+          await dispatch(
+            mmiActions.setCustodianConnectRequest({
+              token: connectRequest.token,
+              apiUrl: connectRequest.apiUrl,
+              custodianName,
+              custodianType: connectRequest.service,
+            }),
+          );
+        }
+
+        await dispatch(
+          mmiActions.removeAddTokenConnectRequest({
+            origin: connectRequest.origin,
+            apiUrl: connectRequest.apiUrl,
+            token: connectRequest.token,
+          }),
+        );
+
+        trackEvent({
+          category: MetaMetricsEventCategory.MMI,
+          event: MetaMetricsEventName.TokenAdded,
+          properties: {
+            actions: isConfirm
+              ? 'Custodian RPC confirm'
+              : 'Custodian RPC cancel',
+            custodian: connectRequest.custodian,
+            apiUrl: connectRequest.apiUrl,
+          },
+        });
+
+        if (isConfirm) {
+          history.push(CUSTODY_ACCOUNT_ROUTE);
+        }
```

Comment:
Missing setLoading false at the end of it?

Reply:
Fixed